src/Entity/Character.cpp

```diff
@@ -96,25 +106,19 @@ void CharacterInput(flecs::entity characterEntity, VelocityComponent& velocityCo
 
 void Attack(flecs::entity& characterEntity)
 {
-	const Vector2 playerCollision = characterEntity.get<CollisionComponent>()->mRectScale;
 	const Vector2 playerPosition = characterEntity.get_ref<TransformComponent>()->mPosition;
-	const Rectangle playerRect{playerPosition.x, playerPosition.y,playerCollision.x, playerCollision.y};
 
 	flecs::world ecs = characterEntity.world();
-	auto filter = ecs.filter_builder<TransformComponent, CollisionComponent, HealthComponent>().with<EnemyTag>().build();
+	auto filter = ecs.filter_builder<TransformComponent, HealthComponent>().with<EnemyTag>().build();
 
-	filter.each([&playerRect](flecs::entity e, TransformComponent& transformComponent, CollisionComponent& collisionComponent, HealthComponent& enemyHealth) {
-		// Define enemy rectangle
-		const Rectangle enemyRect{transformComponent.mPosition.x - 10, transformComponent.mPosition.y - 10, collisionComponent.mRectScale.x + 20, collisionComponent.mRectScale.y + 20};
+	filter.each([&playerPosition, &ecs](flecs::entity e, TransformComponent& transformComponent, HealthComponent& enemyHealth) {
 
-		// Check if position intersects with enemy
-		if (CheckCollisionRecs(playerRect, enemyRect)) {
-			// Deal damage to the enemy
+		if (Vector2Distance(transformComponent.mPosition, playerPosition) <= gAttackRadius){
```

Comment:
Inverting `if` conditions to return early is a great way to reduce nesting, e.g.:
```
if (Vector2Distance(transformComponent.mPosition, playerPosition) > gAttackRadius)
	return;
```